nexus/db-queries/src/db/datastore/bgp.rs

```diff
@@ -492,4 +495,98 @@ impl DataStore {
 
         Ok(results)
     }
+
+    pub async fn communities_for_peer(
+        &self,
+        opctx: &OpContext,
+        port_settings_id: Uuid,
+        interface_name: &String,
+        addr: IpNetwork,
+    ) -> ListResultVec<SwitchPortBgpPeerConfigCommunity> {
+        use db::schema::switch_port_settings_bgp_peer_config_communities::dsl;
+
+        let results = dsl::switch_port_settings_bgp_peer_config_communities
+            .filter(dsl::port_settings_id.eq(port_settings_id))
+            .filter(dsl::interface_name.eq(interface_name.clone()))
+            .filter(dsl::addr.eq(addr))
+            .load_async(&*self.pool_connection_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(results)
+    }
+
+    pub async fn allow_export_for_peer(
+        &self,
+        opctx: &OpContext,
+        port_settings_id: Uuid,
+        interface_name: &String,
+        addr: IpNetwork,
+    ) -> LookupResult<Option<Vec<SwitchPortBgpPeerConfigAllowExport>>> {
+        use db::schema::switch_port_settings_bgp_peer_config as db_peer;
+        use db::schema::switch_port_settings_bgp_peer_config::dsl as peer_dsl;
+        use db::schema::switch_port_settings_bgp_peer_config_allow_export as db_allow;
+        use db::schema::switch_port_settings_bgp_peer_config_allow_export::dsl;
+
+        let active = peer_dsl::switch_port_settings_bgp_peer_config
+            .filter(db_peer::port_settings_id.eq(port_settings_id))
+            .select(db_peer::allow_export_list_active)
+            .limit(1)
+            .first_async::<bool>(
+                &*self.pool_connection_authorized(opctx).await?,
+            )
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        if !active {
```

Comment:
Is there a risk of a TOCTOU here? What happens if it becomes `!active` right after, but before the next query?

Reply:
Put this in a transaction